src/Symfony/Component/Yaml/Parser.php

```diff
@@ -1207,10 +1207,17 @@ private function lexInlineMapping(string $yaml): string
             throw new \InvalidArgumentException(sprintf('"%s" is not a sequence.', $yaml));
         }
 
-        for ($i = 1; isset($yaml[$i]) && '}' !== $yaml[$i]; ++$i) {
+        $openParentheses = 1;
+
+        for ($i = 1; $i < strlen($yaml); ++$i) {
+            if ('{' === $yaml[$i]) {
```

Comment:
be careful. You can have such chars appearing inside quoted strings, in which case it should not be counted in the balancing of parentheses.

Reply:
same for arrays below.

Reply:
Yes, I have pending changes locally for fix the string parsing. They will be part of a different PR as I am not fully done yet.

Reply:
The issue exists in 4.4 already and is just (very) slightly increased. Could be handled in a separate PR before or after as well.
A simple example is:
```yaml
[
"]"
]
```

Reply:
Nitpicking but imo it is even less possible to break with this patch. The brackets are more likely to come in pairs and those break in 4.4 while they are working in this change.
Still a bug surely...

Reply:
@simonberger I have added this one and your other suggestion from below here.